### ISSUE TITLE
Corrupt backups with b2.connections=1

### ISSUE THREAD

**Issue by richard-scott** [restic/restic#1606]
## Output of `restic version`
restic 0.8.1
compiled with go1.9.2 on linux/arm

## How did you run restic exactly?
```
restic --cleanup-cache --option b2.connections=1 backup --exclude-file=/etc/restic/exclude.list --files-from=/etc/restic/include.list
```

## What backend/server/service did you use to store the repository?
Backblaze B2

## Expected behavior
A nice clean backup to be taken.

## Actual behavior

This was the first run of my backup after "restic init" I then ran this:

```
restic --cleanup-cache --option b2.connections=1 check --with-cache
password is correct
create exclusive lock for repository
load indexes
check all packs
pack 10a7aab8a6bd35182854225d9d64ae959fb8dd4ded3cc668f7b966220da1bea2: not referenced in any index
pack 500b6e15d6ab26db1d6c6a0959ecbe30311746e52d3bd4dc447db3a740818b47: not referenced in any index
pack 55ccf4b2119e288b6cdb9ad7e546d93a73239d5b9ec68a36ccaff6a559b9f205: not referenced in any index
check snapshots, trees and blobs
Fatal: repository contains errors
```

## Steps to reproduce the behavior

Backup to Backblaze B2 with --option b2.connections=1.

## Do you have any idea what may have caused this?
Nope.

## Do you have an idea how to solve the issue?
Use --option b2.connections=2

## Did restic help you or made you happy in any way?
Yes, its a fantastic backup tool.

**Comment by richard-scott** [comment]
If I re-run the backup again with b2.connections=2 then the "check" doesn't find any issues, even old issues are non existent.

```
restic --cleanup-cache --option b2.connections=2 check --with-cache
password is correct
create exclusive lock for repository
load indexes
check all packs
check snapshots, trees and blobs
no errors were found
```

**Comment by richard-scott** [comment]
Hmmm, is there a maximum connection limit with B2?

```
restic --cleanup-cache --option b2.connections=20 check --with-cache
create exclusive lock for repository
load indexes
check all packs
pack 1b50bab4da49a834aaa44272393a0a09c4cf3bfe07678f5f51f8ea9f3106633a: not referenced in any index
pack 6c174828995be7130db56c553c8b7faec2830cc19676a323b7b4b7a38a627553: not referenced in any index
check snapshots, trees and blobs
Fatal: repository contains errors
```

**Comment by fd0** [comment]
After 0.8.1 was released, we reworked limiting the backend connection limits. I'm not sure if it works with `connections=1` in 0.8.1. If you like, you can try a beta compiled from the master branch: https://beta.restic.net/?sort=time&order=desc

You may also have experienced the bug described [in the forum](https://forum.restic.net/t/error-loading-tree-check-prune-and-forget-gives-error-b2-backend/406/13) and corrected in #1589. The fix is also not released yet, but included in the latest beta build.

Please report back if you try it!

**Comment by richard-scott** [comment]
Do you have any documents on cross compiling for arm7 on a linux box?
I already have a working gcc-arm-embedded-7-2017q4-1~xenial2 on the box, I'm struggling with the relevant "go run build.go" command.

**Comment by richard-scott** [comment]
It's ok. I've figured it out :-)

```
go run build.go --goos linux --goarch arm
```

**Comment by richard-scott** [comment]
I don't seem to get the errors with a build from "master".

One thing I have noticed is that I don't get the bandwidth speed notifications anymore. Before I used to know what speed I was uploading at, but not anymore.

**Comment by fd0** [comment]
Great, thanks for the feedback. So the issue is solved by #1589, I'm therefore closing it here for now. Please feel free to add further comments!

We've removed the bandwidth display recently, it was very confusing, hard to understand and rarely displayed something relevant. The background here is that it displayed the speed at which the data to be saved is "processed" by restic, which either means at what speed files are read, or (if they already are present in the repo) at which rate restic checks that.

Only after a few minutes during the very first backup into an empty repository did the bandwidth shown correspond roughly to the upload bandwidth.

We'll rework the reporting system once the new archiver code is done. There's an open PR for it.
